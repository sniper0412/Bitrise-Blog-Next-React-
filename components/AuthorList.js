import React from 'react';
import PropTypes from 'prop-types';

import { AuthorType } from '../types';
import { ROUTE_PATHS } from '../config';

const AuthorList = ({ isBanner = false, authors }) => (
  <div className={isBanner ? 'author-header-container' : 'author-info-container'}>
    {isBanner && <div className="all-articles">all articles by</div>}
    <div className="author-info">
      {authors.map(author => (
        <div key={author.slug} className="author-container">
          <img
            alt={`${author.firstName} ${author.lastName} avatar`}
            src={author.profileImage || '/static/svg/default_avatar.svg'}
          />
          <div className="name-and-title">
            <a href={`${ROUTE_PATHS.authors}/${author.slug}`}>
              {author.firstName} {author.lastName}
            </a>
            {author.title && <p>{author.title}</p>}
          </div>
        </div>
      ))}
    </div>
  </div>
);

AuthorList.propTypes = {
  authors: PropTypes.arrayOf(PropTypes.shape(AuthorType)),
  isBanner: PropTypes.bool
};

export default AuthorList;
