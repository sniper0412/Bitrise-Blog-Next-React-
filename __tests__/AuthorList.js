import renderer from 'react-test-renderer';

import AuthorList from '../components/AuthorList';

describe('AuthorList component', () => {
  const authors = [
    {
      slug: 'slug',
      firstName: 'Bender',
      lastName: 'RodrÃ­guez',
      profileImage: 'http://bender.img',
      title: 'Chief Bending Robot',
      bio: 'A fictional character who is one of the main characters in the animated television series Futurama'
    }
  ];

  it('renders correctly', () => {
    const tree = renderer.create(<AuthorList authors={authors} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders correctly in banner mode', () => {
    const tree = renderer.create(<AuthorList authors={authors} isBanner />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
