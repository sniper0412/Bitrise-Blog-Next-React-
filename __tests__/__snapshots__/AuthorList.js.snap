// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AuthorList component renders correctly 1`] = `
<div
  className="author-info-container"
>
  <div
    className="author-info"
  >
    <div
      className="author-container"
    >
      <img
        alt="Bender Rodríguez avatar"
        src="http://bender.img"
      />
      <div
        className="name-and-title"
      >
        <a
          href="/authors/slug"
        >
          Bender
           
          Rodríguez
        </a>
        <p>
          Chief Bending Robot
        </p>
      </div>
    </div>
  </div>
</div>
`;

exports[`AuthorList component renders correctly in banner mode 1`] = `
<div
  className="author-header-container"
>
  <div
    className="all-articles"
  >
    all articles by
  </div>
  <div
    className="author-info"
  >
    <div
      className="author-container"
    >
      <img
        alt="Bender Rodríguez avatar"
        src="http://bender.img"
      />
      <div
        className="name-and-title"
      >
        <a
          href="/authors/slug"
        >
          Bender
           
          Rodríguez
        </a>
        <p>
          Chief Bending Robot
        </p>
      </div>
    </div>
  </div>
</div>
`;
